### DISCUSSION TITLE
How to use dolphin as directory/file picker in gtk applications?

### DISCUSSION THREAD

**Discussion by Gigas002** [hyprwm/Hyprland#4988] [Q&A] [answered]
Hello. Sorry if this has already been asked before, I tried searching for similar issues in `hyprland` and `xdg-desktop-portal-hyprland` repos and didn't found any similar ones.
I want to use `dolphin` as a directory/file picker in firefox, vscode, etc. On a KDE Plasma session it's usually enough to install the `xdg-desktop-portal`, `xdg-desktop-portal-kde` and set the `GTK_USE_PORTAL=1` environment variable and everything will work. However, it doesn't work on `hyprland` session.
What else have I tried:

1. Setting `user_pref`s for firefox:

```js
user_pref("widget.use-xdg-desktop-portal", true);
user_pref("widget.use-xdg-desktop-portal.kde", true);
user_pref("GTK_USE_PORTAL", 1);
user_pref("widget.use-xdg-desktop-portal.file-picker", 1);
```

2. Setting `GDK_DEBUG=portals` environment variable
3. Played with `#gtk-version` flag in `chromium`-based browsers just to make sure it's not GTK version related issue
4. Setting the `XDG_CURRENT_DESKTOP` property to `KDE` or `kwin`

**Comment by Gigas002** [comment, marked as answer]
Alright, got it working. Leaving the solution here for anyone with the same problem. For a reference and a better explanation you can use this article on arch linux wiki: https://wiki.archlinux.org/title/XDG_Desktop_Portal

Create the `~/.config/xdg-desktop-portal` directory with `hyprland-portals.conf` file. File's contents:

```ini
[preferred]
default=hyprland;gtk
org.freedesktop.impl.portal.FileChooser=kde
```

The only addition between default config (located in `/usr/share/xdg-desktop-portal/hyprland-portals.conf`) is the last line which makes it work out.

I think you also need to restart some systemd service in order for it to work with these changes, but I simply rebooted the pc.

**Reply by Rdeisenroth** [reply]
Thanks for sharing this, i created a PR to add it to the Wiki. https://github.com/hyprwm/hyprland-wiki/pull/512

**Comment by erwin** [comment]
Note that to use kde as the FileChooser you probably also want to set `org.freedesktop.FileManager1.service` to dolphin...

This controls which file manager is used to "Show in Folder" for Downloads

Set:

`/usr/share/dbus-1/services/org.freedesktop.FileManager1.service`

```
[D-BUS Service]
Name=org.freedesktop.FileManager1
Exec=/usr/bin/dolphin --no-new-window %u
SystemdService=plasma-dolphin.service
```

**Reply by sjuxax** [reply]
Or symlink to the authoritative version:

```
ln -s /usr/share/dbus-1/services/org.kde.dolphin.FileManager1.service ~/.local/share/dbus-1/services/org.freedesktop.FileManager1.service
```
